swarm/api/api.go

```diff
@@ -202,3 +212,151 @@ func (self *Api) Modify(uri, contentHash, contentType string, nameresolver bool)
 	}
 	return trie.hash.String(), nil
 }
+
+func (self *Api) AddFile(mhash, path, fname string, content []byte, nameresolver bool) (storage.Key, string, error) {
```

Comment:
@lmars That would require lot of changes and i would suggest a different PR for that. In fact i am thinking that the api can be made a little more generic to serve both HTTP and Fuse clients, so that both of them reuse the same function inside api.go. Is that okay?